### PR TITLE
Asynchronous Texture Uploads.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

**Comment by chinmaygarde** [comment]
I am going to hold off on committing this since it is missing backend for multiple platforms.

File: flow/texture_image.cc [hunk 1]
```diff
@@ -31,11 +32,60 @@ inline GLint ToGLDataFormat(TextureImageDataFormat dataFormat) {
...
+    case kRGB_565_SkColorType:
+      dataFormat = TextureImageDataFormat::UnsignedShort565;
+      imageFormat = TextureImageFormat::RGB;
+      break;
+    case kRGBA_8888_SkColorType:
+      dataFormat = TextureImageDataFormat::UnsignedByte;
+      imageFormat = TextureImageFormat::RGBA;
+      break;
+    default:
+      // Add more as supported.
```
Comment (eseidelGoogle) [inline flow/texture_image.cc]: Will users have any way to hit this (based on what kinds of images they provide) or will we only hit this when adding new hardware, or?

Comment (chinmaygarde) [inline flow/texture_image.cc]: Users can definitely hit this. Decode jobs with indexed PNG formats come to mind for example. I am not yet done with the fallback paths. But once that is in place, any cases not caught by this fast path will go down the same old loading routines. Obviously, the more we can catch here, there better for performance. More patches incoming to this PR.

Comment (chinmaygarde) [inline flow/texture_image.cc]: The fallback paths are also necessary for the test targets that don't setup a GL context on the resource loading thread.

File: sky/shell/platform_view.h [hunk 2]
```diff
@@ -56,19 +63,28 @@ class PlatformView {
...
 
   virtual void Resize(const SkISize& size);
 
  protected:
-  explicit PlatformView(const Config& config, SurfaceConfig surface_config);
-
   Config config_;
   SurfaceConfig surface_config_;
+
+  Shell& shell_;
```
Comment (abarth) [inline sky/shell/platform_view.h]: There's no reason to store a reference to the Shell.  It's a global static.

**Comment by abarth** [comment]
LGTM

File: sky/shell/platform_view.h (hunk 2 above)
Comment (chinmaygarde) [inline sky/shell/platform_view.h]: That was in the old shell view that I mushed into the platform view. I think it was there as a way to indicate that an instantiated instance of the shell needed to be provided before a platform view could be created. I am going to get rid of it.

**Comment by chinmaygarde** [comment]
I was getting linker errors related to neon. Upgraded the libpng in third_party to match the one in Skia and unconditionally enable neon optimizations on aarch64 builds (iOS). I have verified that the neon paths are now being hit on iOS and we get a decent reduction in operations on pngs. If you need to verify this for yourself, make sure you set the breakpoints on the renamed symbols in pngprefix.h. These have the skia prefix but we can change this to flutter now if that is cleaner/clearer.

Next up. Android.

**Pull request merged**